### PR TITLE
nixos/castopod: updated nixos test and module

### PR DESCRIPTION
**Pull request by sinavir**
## Description of changes

Made some improvement to help this old pr #260518 to move forward since it is stale because the reviewers don't answer.

- Rebased #260518 on master
- Upgraded castopod
- Added LoadCredential support in the module.
- Reworked a bit the way we generate the mp3file in the test to get rid of the nix-build command inside the test

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).
---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by sinavir** [comment]
Hello, @AleXoundOS @misuzu this is a friendly ping to review the PR and try to move forward for this package that is getting old

**Comment by sinavir** [comment]
Hello @AleXoundOS  @misuzu, this is a friendly ping. Do you think you will have time to review this PR in the next weeks ?

**Comment by sinavir** [comment]
Rebased the PR as someone merged https://github.com/NixOS/nixpkgs/pull/295799

File: nixos/modules/services/web-apps/castopod.nix [hunk 1]
```diff
@@ -111,6 +123,19 @@ in
           Options for Castopod's PHP pool. See the documentation on `php-fpm.conf` for details on configuration directives.
         '';
       };
+      maxUploadSize = lib.mkOption {
+        type = lib.types.int;
+        default = 512;
```
Comment (wegank) [inline nixos/modules/services/web-apps/castopod.nix:128]:
```suggestion
        type = lib.types.str;
        default = "512M";
```

File: nixos/modules/services/web-apps/castopod.nix [hunk 2]
```diff
@@ -142,13 +167,13 @@ in
...
+      inherit phpPackage;
       phpOptions = ''
-        # https://code.castopod.org/adaures/castopod/-/blob/main/docker/production/app/uploads.ini
+        # https://code.castopod.org/adaures/castopod/-/blob/develop/docker/production/common/uploads.template.ini
         file_uploads = On
         memory_limit = 512M
-        upload_max_filesize = 500M
-        post_max_size = 512M
+        upload_max_filesize = ${toString cfg.maxUploadSize}M
+        post_max_size = ${toString cfg.maxUploadSize}M
```
Comment (wegank) [inline nixos/modules/services/web-apps/castopod.nix:176]:
```suggestion
        upload_max_filesize = ${cfg.maxUploadSize}
        post_max_size = ${cfg.maxUploadSize}
```

File: nixos/modules/services/web-apps/castopod.nix [hunk 3]
```diff
@@ -251,6 +281,7 @@ in
         extraConfig = ''
           try_files $uri $uri/ /index.php?$args;
           index index.php index.html;
+          client_max_body_size ${toString cfg.maxUploadSize}M;
```
Comment (wegank) [inline nixos/modules/services/web-apps/castopod.nix:284]:
```suggestion
          client_max_body_size ${cfg.maxUploadSize};
```

File: nixos/modules/services/web-apps/castopod.nix [hunk 4]
```diff
@@ -120,13 +145,13 @@ in
         sslEnabled = with config.services.nginx.virtualHosts.${cfg.localDomain}; addSSL || forceSSL || onlySSL || enableACME || useACMEHost != null;
         baseURL = "http${lib.optionalString sslEnabled "s"}://${cfg.localDomain}";
       in
-      lib.mapAttrs (name: lib.mkDefault) {
+      lib.mapAttrs (_name: lib.mkDefault) {
```
Comment (wegank) [inline nixos/modules/services/web-apps/castopod.nix:148]:
```suggestion
      lib.mapAttrs (_: lib.mkDefault) {
```

**Comment by sinavir** [comment]
Thank you for the review @wegank ! I addressed all your comments

**Comment by wegank** [comment]
@ofborg test castopod

File: nixos/modules/services/web-apps/castopod.nix [hunk 5]
```diff
@@ -278,7 +309,7 @@ in
       };
     };
 
-    users.users.${user} = lib.mapAttrs (name: lib.mkDefault) {
+    users.users.${user} = lib.mapAttrs (_name: lib.mkDefault) {
```
Comment (wegank) [inline nixos/modules/services/web-apps/castopod.nix:312]:
```suggestion
    users.users.${user} = lib.mapAttrs (_: lib.mkDefault) {
```

**Comment by sinavir** [comment]
Tests are passing on my machine

**Comment by wegank** [comment]
Also on ofborg, the aarch64-linux one is not a regression of this PR.

**Review by wegank — APPROVED** [review]

**Pull request merged**